src/components/BulkOperationDetailsPage/BulkOperationDetails.test.jsx

```diff
@@ -0,0 +1,111 @@
+      'b717187c-b8ae-47ac-b65b-ac2274e712db - Intro to Course Loader',
+    ],
+    total_products_count: 2,
+    updated_products_count: 0,
+  },
+};
+
+const csvContent = '"name","image"\n"Test Name","https://example.com/image.jpg"';
+
+describe('BulkOperationDetails', () => {
+  beforeEach(() => {
+    global.fetch = jest.fn().mockResolvedValue({
+      text: jest.fn().mockResolvedValue(csvContent),
+    });
+    render(
+      <IntlProvider locale="en">
+        <BulkOperationDetails task={mockTask} />
+      </IntlProvider>,
+    );
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders task details correctly', () => {
+    expect(screen.getByRole('columnheader', { name: 'Field' })).toBeInTheDocument();
+    expect(screen.getByRole('columnheader', { name: 'Value' })).toBeInTheDocument();
+
+    expect(screen.getByRole('cell', { name: 'Task ID' })).toBeInTheDocument();
+    expect(screen.getByRole('cell', { name: '123' })).toBeInTheDocument();
+
+    expect(screen.getByRole('cell', { name: 'Task type' })).toBeInTheDocument();
+    expect(screen.getByRole('cell', { name: 'course_create' })).toBeInTheDocument();
+
+    expect(screen.getByRole('cell', { name: 'Uploaded by' })).toBeInTheDocument();
+    expect(screen.getByRole('cell', { name: 'test-user' })).toBeInTheDocument();
+
+    expect(screen.getByRole('cell', { name: 'Status' })).toBeInTheDocument();
+    expect(screen.getByRole('cell', { name: 'completed' })).toBeInTheDocument();
+
+    expect(screen.getByRole('cell', { name: 'Created' })).toBeInTheDocument();
+    expect(screen.getByRole('cell', { name: 'Jan 01, 2023, 12:00:00 AM' })).toBeInTheDocument();
+
+    expect(screen.getByRole('cell', { name: 'Modified' })).toBeInTheDocument();
+    expect(screen.getByRole('cell', { name: 'Jan 02, 2023, 12:00:00 AM' })).toBeInTheDocument();
+  });
+
+  it('renders download and preview buttons', () => {
+    const downloadButton = screen.getByRole('link', { name: /Download CSV/i });
+    const previewButton = screen.getByRole('button', { name: /Preview CSV/i });
+
+    expect(downloadButton).toHaveAttribute('href', mockTask.csv_file);
+    expect(downloadButton).toHaveAttribute('target', '_blank');
+    expect(previewButton).toBeInTheDocument();
+  });
+
+  it('fetches and displays CSV content on preview', async () => {
+    const previewButton = screen.getByRole('button', { name: /Preview CSV/i });
+    fireEvent.click(previewButton);
+
+    await waitFor(() => {
+      expect(global.fetch).toHaveBeenCalledWith(mockTask.csv_file);
+    });
+  });
+
+  it('renders BulkOperationDetails and previews CSV', async () => {
+    const previewButton = screen.getByRole('button', { name: /Preview CSV/i });
+    fireEvent.click(previewButton);
+
+    await waitFor(() => {
+      expect(screen.getByRole('dialog')).toBeInTheDocument();
+      const table = screen.getByRole('table');
+      expect(table).toBeInTheDocument();
```

Comment:
nit: should we not assert on parts of content?

Reply:
Sure we can, but there i a sort of dummy data with 2 fields, therefore i didn't add it in the test

Reply:
Let's add the assertions, since the code is previewing CSV and does not care about its structure. The test will verify that preview is working as expected.